tsconfig.json

```
@@ -1,31 +1,24 @@
 {
-    "compilerOptions": {
-      "experimentalDecorators": true,
-      "outDir": "lib",
-      "target": "ESNext",
-      "module": "CommonJS",
-      "moduleResolution": "node",
-      "lib": ["ES2021"],
-      "declaration": true,
-      "skipLibCheck": true,
-      "allowJs": true,
-      "noImplicitAny": true,
-      "esModuleInterop": true,
-      "forceConsistentCasingInFileNames": true,
-      "noFallthroughCasesInSwitch": true,
-      "emitDecoratorMetadata": true,
-      "isolatedModules": true,
-      "resolveJsonModule": true,
-      "downlevelIteration": true,
-      "sourceMap": true
-    },
-    "include": [
-      "src"
-    ],
-    "exclude": [
-      "node_modules",
-      "lib/**/*",
-      "src/**/*.test.ts",
-      "test"
-    ]
-  }
\ No newline at end of file
+  "compilerOptions": {
+    "experimentalDecorators": true,
+    "outDir": "lib",
+    "target": "ESNext",
+    "module": "CommonJS",
+    "moduleResolution": "node",
+    "lib": ["ES2021"],
+    "declaration": true,
+    "skipLibCheck": true,
+    "allowJs": true,
+    "noImplicitAny": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "noFallthroughCasesInSwitch": true,
+    "emitDecoratorMetadata": true,
+    "isolatedModules": true,
+    "resolveJsonModule": true,
+    "downlevelIteration": true,
```

We should add the strict typecheck rule set to the compiler too:
```suggestion
    "downlevelIteration": true,
    "strict": true,
```